### DISCUSSION TITLE
Need to implement API Token like functionality

### DISCUSSION THREAD

**Discussion by ruckc** [keycloak/keycloak#15530] [Q&A]
I need to implement some sort of API Tokens that ideally leverage keycloak.  Ideally, a user is able to create a long lived API token with an expiration.  Any suggestions on how to accomplish this with Keycloak?

We have multiple bearer-only services that it would be great if we could figure out a way to leverage pure keycloak tokens, but that are long-lived and then figuring out how to pick and choose user roles/permissions inside the tokens.

**Comment by ticapix** [comment]
hi @ruckc 

I have the exact same need and would like to leverage keycloak if possible.
Would you share the solution you came up with ?

Best regards,
Pierre

**Reply by ruckc** [reply]
Did it outside of keycloak with custom tokens... I have learned to avoid keycloaknfor anything beyond it's basic capabilities.
